### PR TITLE
Add accessibilityHint for iOS

### PR DESCRIPTION
**Pull request by draperunner**
This adds the accessibilityHint for View, Text and Touchable* on iOS.
The accessibilityHint provides some more information about an element
when the accessibilityLabel is not enough.

## Motivation
The accessibilityHint is a core accessibility property on iOS. 

From https://developer.apple.com/documentation/objectivec/nsobject/1615093-accessibilityhint:
> An accessibility hint helps users understand what will happen when they perform an action on the accessibility element when that result is not obvious from the accessibility label.

Related issue: https://github.com/facebook/react-native/issues/14706

## Test Plan
The npm scripts `test`, `flow`, `lint` and `prettier` are satisfied.

I added a couple of examples to the RNTester app. The Accessibility Inspector on Mac helps debugging accessibility stuff on a simulator, but it does not show the accessibilityHint. Therefore I tested the RNTester app on an iPhone 8 device using VoiceOver to verify the hint functionality. It works fine, and I've tested disabling and enabling "read hints" in the VoiceOver settings on the phone.

## Related PRs
https://github.com/facebook/react-native-website/pull/222

## Release Notes
[IOS][FEATURE][Accessibility] - Add accessibilityHint for View, Text, Touchable* on iOS

### CODE REVIEW

File: React/Views/RCTView.m [hunk 1]
```diff
@@ -154,6 +154,15 @@ - (NSString *)accessibilityLabel
   return RCTRecursiveAccessibilityLabel(self);
 }
 
+- (NSString *)accessibilityHint
```
Comment (janicduplessis) [inline React/Views/RCTView.m]: Why is this needed? Can't we just use the prop directly?

Comment (draperunner) [reply React/Views/RCTView.m]: You're right, it is not needed. I tested after removing it, and it still works. I've updated this in the latest commit, 196ce0658b7c71ec5f30b88018d784b49272cc02

**Comment by janicduplessis** [comment]
Thanks for the PR!

@facebook-github-bot shipit

**Comment by draperunner** [comment]
Hi @janicduplessis, seems like facebook-github-bot met some issues. Anything I can do?

**Comment by hramos** [comment]
@draperunner Janic does not have visibility into the failure but I can help in this regard. It looks like this is held back due to a failing snapshot test. Getting this PR merged will require some manual intervention on my part.

It's possible that the failing test is unrelated to this PR, so let me run the internal diff through the test suite one more time. Please **do not add any more commits** to this PR as doing so will prevent it from landing. Thanks!

**Comment by janicduplessis** [comment]
@hramos This adds a prop to View so it is very possible that some snapshots need to be updated.

**Comment by hramos** [comment]
Indeed that's the case. The failing snapshot tests are all due to `accessibilityHint` being added to the props. I'll take care of this.

**Comment by hramos** [comment]
Snapshot tests updated and test suites are currently running. I'll check again in tomorrow morning and land this if all looks green.

**Comment by hramos** [comment]
```
** BUILD FAILED **
The following commands produced analyzer issues:
Analyze Base/RCTModuleMethod.mm
Analyze Modules/RCTUIManager.m
(2 commands with analyzer issues)
The following build commands failed:
Ld /Users/facebook/Library/Developer/Xcode/DerivedData/RNTester-gmzkrbslacaqklccqmdfkejruuag/Build/Products/Debug-iphonesimulator/RNTester.app/RNTester normal x86_64
```

**Comment by draperunner** [comment]
Any updates on this @hramos?

**Comment by janicduplessis** [comment]
@hramos @draperunner  The PR doesn't touch any of the files from the log, maybe rebase on master and we can try shipping again.

**Comment by draperunner** [comment]
@hramos @janicduplessis Hi guys. I have now rebased this PR and added Flow types.

**Comment by janicduplessis** [comment]
@hramos Wanna try landing this again? It will still require some snapshot updates.

**Comment by knitcodemonkey** [comment]
Cc @thesavior

**Comment by hramos** [comment]
This will be landing soon.
